feat: Import Relay docs

jan-auer opened the pull request:
This imports Relay docs without any changes to their content. Open questions:

- [x] Where should the docs be placed? Now, they live under "Product", but they could also be in their own section, or under Account Management.
- [x] How can we hide them from sidebar navigation? I was able to hide them from indexing, but they still appear on the sidebar for me.
- [x] There are still a few links to old docs.sentry.io pages. Where can we point them now?

In follow-ups, we will:
- Back out rendered metrics docs in favor of auto-generated yml
- Update wording and images
- Update rendering of config and metrics docs

PeloWriter commented on src/docs/product/relay/modes.mdx:
```diff
@@ -0,0 +1,93 @@
+`static`, `proxy`.
+
+Event processing in Sentry can be configured using a two level hierachical
+configuration. At the top level, there are organization wide settings. An
```
This is a bit confusing, so here's a suggested revision:

Event processing in Sentry is configured according to settings defined for the organization and project. Some settings, such as privacy controls, are set at the organization level and inherited by all projects in that organization; other settings are specified per project. For Relay, events are processed according to the project settings to which the event belongs. The mode controls the way Relay obtains these project settings.

PeloWriter commented on src/docs/product/relay/modes.mdx:
```diff
@@ -0,0 +1,93 @@
+mode. If authentication fails, no events will be accepted by Relay.
+
+As Relay receives events from applications, it will request project
+configurations from Sentry in order to process the events. If Sentry is unable
```
```suggestion
configurations from Sentry to process the events. If Sentry is unable
```

PeloWriter commented on src/docs/product/relay/modes.mdx:
```diff
@@ -0,0 +1,93 @@
+As Relay receives events from applications, it will request project
+configurations from Sentry in order to process the events. If Sentry is unable
+to provide the configuration for a particular project, all messages for that
+project will be discared.
```
```suggestion
project will be discarded.
```

PeloWriter commented on src/docs/product/relay/modes.mdx:
````diff
@@ -0,0 +1,93 @@
+```
+
+In static configuration, Relay will process events only for statically
+configured projects and rejects events for all other projects. This is useful
````
```suggestion
configured projects and reject events for all other projects. This is useful
```

PeloWriter commented on src/docs/product/relay/modes.mdx:
```diff
@@ -0,0 +1,93 @@
+
+In `static` mode, Relay does not register with upstream since it does not query
+information from it. After processing events for the configured projects, it
+forwards them to the upstream with the authentication information (DSN) set by
```
```suggestion
forwards them to upstream with the authentication information (DSN) set by
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+## Relay
+
+General relay settings for Relay's operation.
```
```suggestion
General settings for Relay's operation.
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+_String, default: `100MiB`_
+
+The maximum payload size for chunks
```
```suggestion
The maximum payload size for chunks.
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+_Integer, default: number of cpus_
+
+The maximum number of threads to spawn for CPU and web worker, each.
```
```suggestion
The maximum number of threads to spawn for each CPU and web worker.
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+_String, optional_
+
+DSN to report internal Relay failures to.
```
```suggestion
DSN to which to report internal Relay failures.
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+DSN to report internal Relay failures to.
+
+It is not a good idea to set this to a value that will make the Relay send
```
```suggestion
We recommend setting this to a value that will not send Relay errors to itself. Ideally, this value should send errors to Sentry directly, not another Relay.
```

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+DSN to report internal Relay failures to.
+
+It is not a good idea to set this to a value that will make the Relay send
+errors to itself. Ideally this should just send errors to Sentry directly, not
```
remove (incorporated into previous edit)

PeloWriter commented on src/docs/product/relay/options.mdx:
```diff
@@ -0,0 +1,381 @@
+
+It is not a good idea to set this to a value that will make the Relay send
+errors to itself. Ideally this should just send errors to Sentry directly, not
+another Relay.
```
remove (incorporated into previous edit)

PeloWriter commented on src/docs/product/relay/projects.mdx:
```diff
@@ -0,0 +1,122 @@
+
+<Alert level="warning" title="Public Key">
+
+The public key (`<DSN_KEY>`) is the key of the project's DSN and it has nothing
```
```suggestion
The public key (`<DSN_KEY>`) is the key of the project's DSN and has nothing
```

PeloWriter commented on src/docs/product/relay/projects.mdx:
```diff
@@ -0,0 +1,122 @@
+<Alert level="warning" title="Public Key">
+
+The public key (`<DSN_KEY>`) is the key of the project's DSN and it has nothing
+to do with the Relay public key that is used for Relay registration.
```
```suggestion
to do with the Relay public key used for Relay registration.
```

PeloWriter commented on src/docs/product/relay/projects.mdx:
````diff
@@ -0,0 +1,122 @@
+12345abcdb1e4c123490ecec89c1f199
+```
+
+A project may contain multiple public keys, only messages using enabled project
````
```suggestion
A project may contain multiple public keys, but only messages using enabled project
```

PeloWriter reviewed (commented):
copy edits primarily, with one question on settings - I've suggested a revision.

jan-auer commented:
Thanks, @PeloWriter, for the thorough review! To keep the documentation in sync with the Relay repo until we remove it from there, I'll apply your suggestions in a follow-up PR. 

- Fixed platform redirects via https://github.com/getsentry/sentry-docs/pull/2197/commits/2f77912415929d513250b907f077f3ad99deffb1. The message on the redirect page is not ideal as it suggests the content has "moved", but it is clear to use.
- Excluded Relay from the sidebar via https://github.com/getsentry/sentry-docs/pull/2197/commits/9a5bc14ad3f4e3b1ccf3bc5f36b86591f4aafea8 and added a page grid instead.

jan-auer commented on src/docs/product/relay/metrics.mdx:
```diff
@@ -0,0 +1,386 @@
+Counts the number of messages placed on the Kafka queue.
+
+When Relay operates with processing enabled and an item is successfully processed, each item
+will generate a message on the Kafka. The counter has an `event_type` tag which is set to
```
Updated in Relay.
